docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/CgovCoreTwigExtensions.php

```diff
@@ -0,0 +1,170 @@
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current Request object.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   */
+  protected $requestStack;
+
+  /**
+   * Constructs a new CgovNavigationManager class.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, RequestStack $requestStack) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->requestStack = $requestStack;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getName() {
+    return 'cgov_core.CgovCoreTwigExtensions';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFunctions() {
+    return [
+      new \Twig_SimpleFunction('get_enclosure', [$this, 'getEnclosure'], ['is_safe' => ['html']]),
+      new \Twig_SimpleFunction('get_list_description', [$this, 'getListDescription'], ['is_safe' => ['html']]),
+    ];
+  }
+
+  /**
+   * Generate <enclosure url='x' length='9' type='mime/type' /> tag from NID.
+   *
+   * Call this function with {{ get_enclosure(nid) }}
+   *
+   * @param int $nid
+   *   Node ID to create enclosure tag from.
+   *
+   * @return string
+   *   generated enclosure tag.
+   */
+  public function getEnclosure($nid) {
+    // Set $debug=TRUE to show displayed image source in comments.
+    $debug = FALSE;
+
+    // Image style to use when rendering images.
+    $thumbnail_image_style = 'cgov_thumbnail';
+
+    // Load the node.
+    $node = $this->entityTypeManager->getStorage('node')->load($nid);
+    if (!$node) {
+      return FALSE;
+    }
+
+    // Check for available images to display.
+    if ($node->hasField('field_image_article')) {
+      $displaying = 'article';
+      $media_field = $node->get('field_image_article')->entity;
+      $media_image_file = $media_field->get('field_media_image')->entity;
+      // Check for overridden thumbnail image.
+      $media_image_thumbnail_file = $media_field->get('field_override_img_thumbnail')->entity;
+      if ($media_image_thumbnail_file) {
+        $displaying = 'article thumbnail override';
+        $media_image_file = $media_image_thumbnail_file;
+      }
+    }
+
+    // Check if there is a promotional image specified, if so use it.
+    if ($node->hasField('field_image_promotional')) {
+      $promo_media_field = $node->get('field_image_promotional')->entity;
+      // Check if promo image is present.
+      if ($promo_media_field) {
+        $displaying = 'promo';
+        $media_image_file = $promo_media_field->get('field_media_image')->entity;
+
+        // Check for overridden thumbnail image.
+        $media_image_thumbnail_file = $promo_media_field->get('field_override_img_thumbnail')->entity;
+        if ($media_image_thumbnail_file) {
+          $displaying = 'promo thumbnail override';
+          $media_image_file = $media_image_thumbnail_file;
+        }
+      }
+    }
+
+    // Get URI of media image file (eg: public://2019-03/image.jpg).
+    $image_uri = $media_image_file->getFileUri();
```

Comment:
Can you add a null check for `$media_image_file` here?

Reply:
It should never be null, as the file field is required and it has already checked for the presence of the Image media entity above.